corehq/apps/locations/bulk_management.py

```diff
@@ -326,10 +346,11 @@ def validate_and_parse_stubs_from_excel(self):
 
         # all locations sheets should have correct headers
         location_stubs = []
+        optional_headers = [LOCATION_SHEET_HEADERS['custom_data'], LOCATION_SHEET_HEADERS['uncategorized_data']]
         for sheet_name, sheet_reader in sheets_by_title.items():
             if sheet_name != self.types_sheet_title:
-                actual = set(sheet_reader.fieldnames)
-                expected = set(LOCATION_SHEET_HEADERS.values())
+                actual = set(sheet_reader.fieldnames) - set(optional_headers)
+                expected = set(LOCATION_SHEET_HEADERS.values()) - set(optional_headers)
```

Comment:
I think these aren't necessarily all optional, you can specify some fields as required (see https://www.commcarehq.org/a/ethan-commtrack/settings/locations/fields/ for example).  It's probably fine to skip those titles for the header validation (as you've done here), but then I think we need the `CustomDataEditor.is_valid` thing to make sure that users aren't uploading invalid custom data.

I'd also be okay with skipping custom data stuff altogether for this first pass and just saying we don't yet support it through upload, then getting that fix in separately.

Sorry, I should've caught this in an earlier pass, but I didn't think of it.